### PR TITLE
Migrate flutter_cache to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/lib/src/flutter_cache.dart [hunk 1]
```diff
@@ -139,9 +137,12 @@ class MaterialFonts extends CachedArtifact {
     ArtifactUpdater artifactUpdater,
     FileSystem fileSystem,
     OperatingSystemUtils operatingSystemUtils,
-  ) {
-    final Uri archiveUri = _toStorageUri(version);
-    return artifactUpdater.downloadZipArchive('Downloading Material fonts...', archiveUri, location);
+  ) async {
+    final String? cacheVersion = version;
+    if (cacheVersion != null) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/flutter_cache.dart:142]: Not sure if there's anything to be done if the version can't be parsed, it should always be present at this point.

File: packages/flutter_tools/lib/src/flutter_cache.dart [hunk 2]
```diff
@@ -537,14 +538,18 @@ class GradleWrapper extends CachedArtifact {
...
-    final Uri archiveUri = _toStorageUri(version);
-    await artifactUpdater.downloadZippedTarball('Downloading Gradle Wrapper...', archiveUri, location);
-    // Delete property file, allowing templates to provide it.
-    // Remove NOTICE file. Should not be part of the template.
-    final File propertiesFile = fileSystem.file(fileSystem.path.join(location.path, 'gradle', 'wrapper', 'gradle-wrapper.properties'));
-    final File noticeFile = fileSystem.file(fileSystem.path.join(location.path, 'NOTICE'));
-    ErrorHandlingFileSystem.deleteIfExists(propertiesFile);
-    ErrorHandlingFileSystem.deleteIfExists(noticeFile);
+    final String? cacheVersion = version;
+    if (cacheVersion != null) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/flutter_cache.dart:542]: Same.

File: packages/flutter_tools/lib/src/flutter_cache.dart [hunk 3]
```diff
@@ -662,16 +667,19 @@ class FlutterRunnerDebugSymbols extends CachedArtifact {
...
   Future<void> _downloadDebugSymbols(String targetArch, ArtifactUpdater artifactUpdater) async {
-    final String packageName = 'fuchsia-debug-symbols-$targetArch';
-    final String url = packageResolver.resolveUrl(packageName, version);
-    await artifactUpdater.downloadZipArchive(
-      'Downloading debug symbols for flutter runner - arch:$targetArch...',
-      Uri.parse(url),
-      location,
-    );
+    final String? cacheVersion = version;
+    if (cacheVersion != null) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/flutter_cache.dart:674]: Same.

File: packages/flutter_tools/lib/src/flutter_cache.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/flutter_cache.dart:142]: I think the right thing to do is `!` so if it does fail it shows up in crash logging

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/lib/src/flutter_cache.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/flutter_cache.dart]: Done, had to update an additional unit test.

**Pull request merged**